File: api/tv.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import requests

app = Flask(__name__)
CORS(app)  # Enable CORS for all routes

@app.route('/tv', methods=['GET'])
def get_tv():
    search = request.args.get('search', '')
    if not search:
        return jsonify({"error": "Missing search parameter"}), 400

    # Original API URL
    original_api = f"https://tv-chi-eosin.vercel.app/tv?search={search}"
    try:
        response = requests.get(original_api)
        data = response.json()
    except Exception as e:
        return jsonify({"error": str(e)}), 500

    return jsonify(data)

# For Vercel deployment
def handler(request, context):
    return app(request.environ, context)

File: api/tv.js
import fetch from "node-fetch";

export default async function handler(req, res) {
  const search = req.query.search || "";

  try {
    const response = await fetch(`https://tv-chi-eosin.vercel.app/tv?search=${encodeURIComponent(search)}`);
    const data = await response.json();

    // Enable CORS
    res.setHeader("Access-Control-Allow-Origin", "*");
    res.setHeader("Access-Control-Allow-Methods", "GET,OPTIONS");
    res.setHeader("Access-Control-Allow-Headers", "Content-Type");

    // Handle preflight
    if (req.method === "OPTIONS") {
      res.status(200).end();
      return;
    }

    res.status(200).json(data);
  } catch (err) {
    res.status(500).json({ error: "Failed to fetch data" });
  }
}
